Modularizing, adding prerequisites. (#1484)
codenvy/che-docs: src/main/pages/che-7/installation-guide/proc_configuring-che-to-run-in-a-restricted-environment.adoc

[id="configuring-che-to-run-in-a-restricted-environment_{context}"]
= Configuring {prod-short} to run in a restricted environment

This section describes how to configure {prod-short} and related containers to use images in a restricted environment rather than the default images in an online registry.

.Prerequisites

> * All required images available in an image registry that is visible to the cluster where {prod-short} is to be deployed.

Reviewer: ```suggestion
* All required images available in an image registry that is accessible from the cluster where {prod-short} is to be deployed.
```

.Procedure

This procedure uses the following placeholders:

.Placeholders used in examples
[cols="1,2"]
|===
| `<my-internal-registry>`
| a host name of the container-image registry (which is accessible in the restricted environment, see xref:pulling-images-used-by-{prod-id-short}-to-a-private-registry_{context}[])

| `<my-organization>`
| organization of the container-image registry

| `<my-offline-devfile-registry>` +
  `<my-offline-plug-in-registry>`
| offline plug-in and devfile registries (see xref:building-offline-versions-of-the-plug-in-and-devfile-registry_{context}[])

| `<ver>`
| release of {prod-short} that is being deployed
|===

> The `CheCluster` Custom Resource, which is managed by the {prod-short} Operator, includes fields to facilitate deploying an instance of {prod-short} in a restricted environment:

Reviewer: ```suggestion
The `CheCluster` custom resource, which is managed by the {prod-short} Operator, includes fields to facilitate deploying an instance of {prod-short} in a restricted environment:
```
https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/

[source,yaml,subs="+quotes"]
----
# [...]
spec:
  server:
    airGapContainerRegistryHostname: '__<my-internal-registry>__'
    airGapContainerRegistryOrganization: '__<my-organization>__'
# [...]
----

Setting these values uses `<my-internal-registry>` and `<my-organization>` for all images. This means that the Operator expects the offline plug-in and devfile registries to be available at:

[subs="+quotes"]
----

> __<my-internal-registry>__/__<my-organization>__/my-offline-plug-in-registry:__<ver>__

Reviewer: ```suggestion
__<my-internal-registry>__/__<my-organization>__/__<my-offline-plug-in-registry>__:__<ver>__
```

> __<my-internal-registry>__/__<my-organization>__/my-offline-plug-in-registry:__<ver>__

Reviewer: ```suggestion
__<my-internal-registry>__/__<my-organization>__/__<my-offline-devfile-registry>__:__<ver>__
```
